File: src/ios/OSCheckAppPlugin.m
/********* OSCheckAppPlugin.m Cordova Plugin Implementation *******/

#import <Cordova/CDV.h>

@interface OSCheckAppPlugin : CDVPlugin {
  // Member variables go here.
}

- (void)isAppInstalled:(CDVInvokedUrlCommand*)command;
@end

@implementation OSCheckAppPlugin

- (void)isAppInstalled:(CDVInvokedUrlCommand*)command
{
    NSString* urlScheme = [command.arguments objectAtIndex:0];

    if([[UIApplication sharedApplication] canOpenUrl:[NSURL URLWithString:urlScheme]]){
        CDVPluginResult* pluginResult = [CDVPluginResult CDVPluginResult: CDVCommandStatus_OK messageAsInt: 1];
        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    }else{
        CDVPluginResult* pluginResult = [CDVPluginResult CDVPluginResult: CDVCommandStatus_OK messageAsInt: 0];
        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    }
}

@end

File: src/ios/OSCheckAppPlugin.swift
import Foundation
import UIKit

@objc(OSCheckAppPlugin) class OSCheckAppPlugin: CDVPlugin {
 @objc(checkAppInstalled:)
 func checkAppInstalled(command: CDVInvokedUrlCommand) {
   guard let scheme = command.arguments[0] as? String,
         let url = URL(string: "\(scheme)://") else {
     self.commandDelegate.send(CDVPluginResult(status: .error), callbackId: command.callbackId)
     return
   }
   let canOpen = UIApplication.shared.isAppInstalled(url)
   let result = CDVPluginResult(status: .ok, messageAs: canOpen ? 1 : 0)
   self.commandDelegate.send(result, callbackId: command.callbackId)
 }

 @objc(openApp:)
 func openApp(command: CDVInvokedUrlCommand) {
   guard let scheme = command.arguments[0] as? String,
         let url = URL(string: "\(scheme)://"),
         UIApplication.shared.isAppInstalled(url) else {
     self.commandDelegate.send(CDVPluginResult(status: .error, messageAs: "App not found"), callbackId: command.callbackId)
     return
   }

   UIApplication.shared.open(url, options: [:]) { success in
     let result = CDVPluginResult(status: .ok, messageAs: "Opened app")
     self.commandDelegate.send(result, callbackId: command.callbackId)
   }
 }
}
